src/common/hist_util.cc

```diff
@@ -534,31 +660,23 @@ FindGroups(const std::vector<unsigned>& feature_list,
       search_groups.resize(param.max_search_group);
     }
 
-    // examine each candidate group: is it okay to insert fid?
-    for (auto gid : search_groups) {
-      const size_t rest_max_cnt = max_conflict_cnt - group_conflict_cnt[gid];
-      const size_t cnt = GetConflictCount(conflict_marks[gid], column, rest_max_cnt);
-      if (cnt <= rest_max_cnt) {
-        need_new_group = false;
-        groups[gid].push_back(fid);
-        group_conflict_cnt[gid] += cnt;
-        group_nnz[gid] += cur_fid_nnz - cnt;
-        MarkUsed(&conflict_marks[gid], column);
+    switch (colmat.GetTypeSize()) {
```

Comment:
Why UINT8_BINS_TYPE-like enam's are not used here?
It will be more consistent approach and more readable.

Reply:
UINT8_BINS_TYPE was renamed to UINT8_BINS_TYPE_SIZE, and applied to column also :)